### PR TITLE
[Merged by Bors] - feat: port data.char

### PR DESCRIPTION
**Pull request by arienmalec**
mathlib3 SHA: aab56fd783b265168f660d6b46709961153c4c20

Note: This file already existed in Mathlib4, so the material from mathlib3 was just added, but didn't replace the existing results.

### CODE REVIEW

File: Mathlib/Data/Char.lean [hunk 1]
```diff
@@ -1,3 +1,18 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import Mathlib.Data.UInt
+import Mathlib.Init.Algebra.Order
+
+/-!
+# More `char` instances
```
Comment (semorrison) [inline Mathlib/Data/Char.lean:10]:
```suggestion
# More `Char` instances
```

**Comment by semorrison** [comment]
bors d+

**Comment by j-loreaux** [comment]
maintainer merge

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**